pyproject.toml

```diff
@@ -2,6 +2,80 @@
 requires = ["setuptools >= 40.6.0", "setuptools_scm[toml] >=4,<6.1", "cython<3", "cmake"]
 build-backend = "setuptools.build_meta"
 
+[project]
```

Comment:
This appears to duplicate a bunch of information from [setup.py](https://github.com/DataDog/dd-trace-py/blob/a67eb999fc04a48730079edfe42bd01c03daf112/setup.py#L503). If possible, let's make it a move instead of a copy, and remove these data from setup.py.